highlights.lua

```diff
@@ -22,6 +22,26 @@ end
 
 --- @param doc core.doc
 function M.init(doc)
+	local function getSource(n)
+		local startPt = n:start_point()
+		local endPt   = n:end_point()
+		local startRow, startCol = startPt.row + 1, startPt.column + 1
+		local endRow, endCol     = endPt.row + 1, endPt.column
+
+		if startRow == endRow then
+			return doc.lines[startRow]:sub(startCol, endCol)
+		end
+
+		local lns = {}
+		lns[1] = doc.lines[startRow]:sub(startCol)
+		for i = startRow + 1, endRow - 1 do
+			lns[#lns + 1] = doc.lines[i]
+		end
+		lns[#lns + 1] = doc.lines[endRow]:sub(1, endCol)
+
+		return table.concat(lns)
+	end
+
```

Comment:
i wonder if this is slower than getting it from treesitter on long lines

Reply:
i would rather just wait for a fix from ltreesitter if it is significantly worse

Reply:
Have not tested it, but it could go either way. IIRC Lua heavily optimises string handling, and passing strings in from C might not be much faster, or even just slower than concatenating strings in Lua itself.

I'll update when I have the time.